### PR TITLE
Added a section about use cases for lambda expressions.

### PR DESCRIPTION
**Pull request by dasdachs**
I opened #3372 during Hacktoberfest 2017, but when a prior contribution was merged to `master`, my PR had unresolved conflict issues. I resolved the conflicts and combined both entries.

### CODE REVIEW

**Comment by thomaskuntzz** [comment]
There is also #3526 for this article

**Comment by dasdachs** [comment]
@thomaskuntzz I didn't notice. Lambda expressions are unimaginably popular :)

Do you happen to know if #3526 is done? If so, I can rewrite my PR. Otherwise I suggest that we wait to see which gets pulled first and rewrite accordingly.

**Comment by thomaskuntzz** [comment]
Yes, #3526 is ready to merge 🙂

**Comment by thomaskuntzz** [comment]
I can merge it now if you want, so that it'll be easier for you to solve merge conflicts

**Comment by dasdachs** [comment]
@thomaskuntzz that would be great, thank you! I'll update my PR in the evening.

**Comment by thomaskuntzz** [comment]
Cool, thanks a lot for your contribution! 👍

**Comment by dasdachs** [comment]
Thank you for your work with the guides!

**Comment by nsuchy** [comment]
@thomaskuntzz What's the status on this PR? Is it merge ready?

**Comment by nsuchy** [comment]
@thomaskuntzz Can you review this?

**Comment by dasdachs** [comment]
@thomaskuntzz as far as I can tell this is from pull [#2077](https://github.com/freeCodeCamp/guide/pull/2077) taken from Python's [documentation](https://docs.python.org/3/reference/expressions.html?highlight=lambda#lambda):

> Lambda expressions (sometimes called lambda forms) are used to create anonymous functions. The expression lambda parameters: expression **yields** a function object.

IMHO this is a valid explanation.

**Comment by nsuchy** [comment]
@thomaskuntzz The documentation backing it is linked in resources, I think this is good to merge. Any further changes to request or am I good to merge?

File: ** 2` yields an anonymous function object which can be associated with any name. [hunk 1]
```diff
@@ -0,0 +1,7 @@
+" Press ? for help
```
[the remark this answers is not in the document]
Comment (nsuchy) [reply ** 2` yields an anonymous function object which can be associated with any name.]: A .swp file also got included which doesn't need to be here.

[the remark this answers is not in the document]
Comment (dasdachs) [reply ** 2` yields an anonymous function object which can be associated with any name.]: Sorry for that. TBH I have no idea how I included a .swp file and overlooked it.

**Comment by nsuchy** [comment]
@thomaskuntzz Well this was challenging to review and get changed but glad to see we saw it to the end and got this merged 😄

**Comment by dasdachs** [comment]
@nsuchy @thomaskuntzz thanks guys!

**Pull request merged**